cirq/contrib/routing/greedy.py

```diff
@@ -284,7 +284,9 @@ def apply_next_swaps(self):
                 ]
                 if len(candidate_swap_sets) == 1:
                     self.apply_swap(*candidate_swap_sets[0])
-                    return
+
+                    if len(list(self.remaining_dag.findall_nodes_until_blocked(self.acts_on_nonadjacent_qubits))) > 0:
```

Comment:
```suggestion
                    if list(self.remaining_dag.findall_nodes_until_blocked(self.acts_on_nonadjacent_qubits))):
                        return
                    else:
                        break               
```

Reply:
No reason to keep going through the time slices.

Reply:
You're right, fixed.